File: freecodecamp/bonfire/13-mutations.js
function mutation(arr) {
  function dedupe(arr) {
    return arr.filter(function(item, position) {
      return arr.indexOf(item) == position;
    });
  }

  var dedupedTarget = dedupe(arr[0].split('')).join('');
  var dedupedPattern = dedupe(arr[1].toLowerCase().split('')).join('');
  var re = new RegExp('[' + dedupedPattern + ']', 'gi');

  return dedupedTarget.match(re).length == dedupedPattern.length ? true : false;
}

console.log( mutation(['hello', 'hey']));
console.log( mutation(['hello', 'Hello']));
console.log( mutation(['zyxwvutsrqponmlkjihgfedcba', 'qrstu']));
console.log( mutation(['Mary', 'Army']));
console.log( mutation(['Alien', 'line']));
console.log( mutation(['floor', 'for']));
console.log( mutation(['hello', 'neo']));

File: freecodecamp/bonfire/13-mutations.rb
def mutation(array)
  array.last.downcase.split('').map { |letter| array.first.downcase.split('').include? letter }.uniq.length == 1 ? true : false
end

p mutation(['hello', 'hey'])
p mutation(['hello', 'Hello'])
p mutation(['zyxwvutsrqponmlkjihgfedcba', 'qrstu'])
p mutation(['Mary', 'Army'])
p mutation(['Alien', 'line'])
p mutation(['floor', 'for'])
p mutation(['hello', 'neo'])
